internal/pkg/admission/scheduler/handler.go

```diff
@@ -36,6 +37,20 @@ var _ admission.Handler = &PodSchedulerSetter{}
 
 // Handle handles an admission request and mutates a pod object in the request.
 func (p *PodSchedulerSetter) Handle(ctx context.Context, req types.Request) types.Response {
+	// NOTE: Since this webhook handler is built from an internal copy of
+	// controller-runtime webhook, the webhook admission decoder in the
+	// controller manager doesn't get translated to the internal webhook
+	// admission decoder correctly. As a workaround, since this handler is for
+	// pods only, use client-go's default scheme to create a new decoder.
+	// Client-go's default scheme contains pod scheme.
+	if p.decoder == nil {
+		decoder, err := admission.NewDecoder(kscheme.Scheme)
+		if err != nil {
+			log.Printf("failed to create a new decoder for the webhook handler: %v", err)
+			return admission.ErrorResponse(http.StatusInternalServerError, err)
+		}
+		p.decoder = decoder
+	}
```

Comment:
Have to use a custom decoder here because the webhook handler's decoder remains nil when mixing the webhook admission types (original and copy). The controller manager gets initialized with the default decoder for webhook handlers but it fails to propagate to the webhook created by the internal copy of the webhook admission package.
Without this, the mutation results in panic:
```
2019/12/03 21:58:56 http: panic serving 10.244.0.1:43920: runtime error: invalid memory address or nil pointer dereference
goroutine 1029 [running]:
net/http.(*conn).serve.func1(0xc0003c1680)
	/usr/local/go/src/net/http/server.go:1769 +0x139
panic(0x132f2e0, 0x2330d30)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/storageos/cluster-operator/internal/pkg/admission/scheduler.(*PodSchedulerSetter).Handle(0xc00078ed80, 0x16c8ca0, 0xc000042030, 0xc0008331e0, 0x12351d8, 0xf, 0xffffffffffffffff, 0xc0008419d8)
	/go/src/github.com/storageos/cluster-operator/internal/pkg/admission/scheduler/handler.go:41 +0x59
github.com/storageos/cluster-operator/internal/pkg/crv01/webhook/admission.(*Webhook).handleMutating(0xc00035a100, 0x16c8ca0, 0xc000042030, 0xc0008331e0, 0xf98, 0xc00025ae40, 0xc00025ae40, 0x0)
	/go/src/github.com/storageos/cluster-operator/internal/pkg/crv01/webhook/admission/webhook.go:134 +0xee
github.com/storageos/cluster-operator/internal/pkg/crv01/webhook/admission.(*Webhook).Handle(0xc00035a100, 0x16c8ca0, 0xc000042030, 0xc0008331e0, 0x0, 0x16a8de0, 0xc000d69890, 0x16a8de0)
	/go/src/github.com/storageos/cluster-operator/internal/pkg/crv01/webhook/admission/webhook.go:121 +0x220
github.com/storageos/cluster-operator/internal/pkg/crv01/webhook/admission.(*Webhook).ServeHTTP(0xc00035a100, 0x16bf960, 0xc000df61c0, 0xc0004e0800)
	/go/src/github.com/storageos/cluster-operator/internal/pkg/crv01/webhook/admission/http.go:94 +0x9cf
```